code/modules/food&drinks/drinks/drinks.dm

```diff
@@ -288,3 +298,302 @@
+		reagents.add_reagent("nicotine", 8)
+		reagents.add_reagent("????", 10)
+		reagents.add_reagent("potassium", 5)
+
+/obj/item/weapon/reagent_containers/food/drinks/dansoup/mushroom
+	name = "Frycook Dan's Quik-Noodles - Mushroom-Swiss Burger-Bake Flavor"
+	New()
+		..()
+		reagents.add_reagent("chicken_soup", 10)
+		reagents.add_reagent("soybeanoil", 1)
+		reagents.add_reagent("msg", 9)
+		reagents.add_reagent("sodiumchloride", 10)
+		reagents.add_reagent("nicotine", 8)
+		reagents.add_reagent("beff", 2)
+		reagents.add_reagent("weird_cheese", 2)
+		reagents.add_reagent("psilocybin", 6)
+
+/obj/item/weapon/reagent_containers/food/drinks/dansoup/ketchup
+	name = "Frycook Dan's Quik-Noodles - Curly Fry Ketchup Hoedown Flavor"
+	New()
+		..()
+		reagents.add_reagent("chicken_soup", 10)
+		reagents.add_reagent("hydrogenated_soybeanoil", 1)
+		reagents.add_reagent("msg", 9)
+		reagents.add_reagent("sodiumchloride", 10)
+		reagents.add_reagent("nicotine", 8)
+		reagents.add_reagent("tomatojuice", 4)
+		reagents.add_reagent("mugwort", 3)
+		reagents.add_reagent("capsaicin", 3)
+
+/obj/item/weapon/reagent_containers/food/drinks/dansoup/sundae
+	name = "Dessert Dan's Quik-Noodles - Sweet Sundae Noodle Flavor"
+	New()
+		..()
+		reagents.add_reagent("chicken_soup", 10)
+		reagents.add_reagent("soybeanoil", 1)
+		reagents.add_reagent("msg", 9)
+		reagents.add_reagent("sodiumchloride", 10)
+		reagents.add_reagent("nicotine", 8)
+		reagents.add_reagent("vhfcs", 10)
+		reagents.add_reagent("chocolate", 2)
+		reagents.add_reagent("cream", 2)
+		reagents.add_reagent("chocolate_milk", 6)
+
+/obj/item/weapon/reagent_containers/food/drinks/dansoup/tuna
+	name = "Descuento Danito's Quik-Noodles - Tuna Melt Taco Fiesta Flavor"
+	New()
+		..()
+		reagents.add_reagent("chicken_soup", 10)
+		reagents.add_reagent("soybeanoil", 1)
+		reagents.add_reagent("msg", 9)
+		reagents.add_reagent("sodiumchloride", 10)
+		reagents.add_reagent("nicotine", 8)
+		reagents.add_reagent("fake_cheese", 3)
+		reagents.add_reagent("mercury", 3)
+		reagents.add_reagent("capsaicin", 4)
+
+/obj/item/weapon/reagent_containers/food/drinks/dansoup/lomein
+	name = "Discount Deng's Quik-Noodles - Sweet and Sour Lo Mein Flavor"
+	New()
+		..()
+		reagents.add_reagent("chicken_soup", 10)
+		reagents.add_reagent("soybeanoil", 1)
+		reagents.add_reagent("msg", 9)
+		reagents.add_reagent("sodiumchloride", 10)
+		reagents.add_reagent("nicotine", 8)
+		reagents.add_reagent("sacid", 3)
+		reagents.add_reagent("vhfcs", 7)
+
+/obj/item/weapon/reagent_containers/food/drinks/dansoup/crab
+	name = "Pirate Dan's Quik-Noodles - Spicy Imitation Crab Meat Paste Flavor"
+	New()
+		..()
+		reagents.add_reagent("chicken_soup", 10)
+		reagents.add_reagent("soybeanoil", 1)
+		reagents.add_reagent("msg", 9)
+		reagents.add_reagent("sodiumchloride", 10)
+		reagents.add_reagent("nicotine", 8)
+		reagents.add_reagent("synthflesh", 3)
+		reagents.add_reagent("saltpetre", 3)
+		reagents.add_reagent("capsaicin", 14)
```

Comment:
taking a meal from a dispenser and putting booze in it shouldn't yield anti-riot pepper spray.